lib/node/Type.js

```diff
@@ -96,32 +96,44 @@ NodeType.isIntegerFraction = function(node, allowUnaryMinus=false) {
           Number.isInteger(parseFloat(denominator.value)));
 };
 
-NodeType.isFraction = function(node, allowUnaryMinus=true, allowParens=true) {
-  if (NodeType.isOperator(node, '/')) {
+// Returns true if `node` belongs to the type specified by boolean `isTypeFunc`.
+// If `allowParens` is true, we allow for the node to be nested within parenthesis.
+NodeType.isType = function(node, isTypeFunc, allowParens=true) {
+  if (isTypeFunc(node)) {
     return true;
   }
-  else if (allowUnaryMinus && NodeType.isUnaryMinus(node)) {
-    return NodeType.isFraction(node.args[0]);
-  }
   else if (allowParens && NodeType.isParenthesis(node)) {
-    return NodeType.isFraction(node.content);
+    return NodeType.isType(node.content, isTypeFunc, allowParens);
   }
 
   return false;
 };
 
-NodeType.getFraction = function(node, allowUnaryMinus=true, allowParens=true) {
-  if (NodeType.isOperator(node, '/')) {
+// Returns `node` if `node` belongs to the type specified by boolean `isTypeFunc`.
+// If `allowParens` is true, we check for an inner node of this type.
+NodeType.getType = function(node, isTypeFunc, allowParens=true) {
+  if (isTypeFunc(node)) {
     return node;
   }
-  else if (allowUnaryMinus && NodeType.isUnaryMinus(node)) {
-    return NodeType.getFraction(node.args[0]);
-  }
   else if (allowParens && NodeType.isParenthesis(node)) {
-    return NodeType.getFraction(node.content);
+    return NodeType.getType(node.content, isTypeFunc, allowParens);
   }
 
-  throw Error('Expected fraction');
+  throw Error('`getType` called on a node that does not belong to specified type');
+};
+
+NodeType.isFraction = function(node, allowParens=true) {
+  return NodeType.isType(
+    node,
+    function(node) { return NodeType.isOperator(node, '/'); },
```

Comment:
in es6 I think this is generally done `(node) => NodeType.isOperator(node, '/')`